packages/flutter/lib/src/widgets/drag_target.dart

```diff
@@ -418,26 +422,30 @@ class _DragAvatar<T> extends Drag {
     _entry.markNeedsBuild();
     HitTestResult result = new HitTestResult();
     WidgetFlutterBinding.instance.hitTest(result, globalPosition + feedbackOffset);
-    _DragTargetState<T> target = _getDragTarget(result.path);
-    if (target == _activeTarget)
+    List<_DragTargetState<T>> targets = _getDragTargets(result.path);
+    if (targets.contains(_activeTarget))
```

Comment:
this is wrong. You need to check that it's still the one that would be selected. Or, track the list you had last time and make sure it's the same this time.